Framework/PythonInterface/plugins/algorithms/PeakMatching.py

```diff
@@ -0,0 +1,307 @@
+            peaks.append(table_dict[centre_column_name][i])
+            sigma.append(table_dict[sigma_column_name][i])
+
+        return list(zip(peaks, sigma))
+
+    def get_matches(self,peak_data, input_peaks):
+        all_matches = []
+        for x in peak_data:
+            raw_data = peak_data[x]
+            matches = []
+            for peak, sigma in input_peaks:
+                for element in raw_data:
+                    for transition, energy in raw_data[element].items():
+
+                        if peak == energy:
+                            data = {}
+                            data['element'] = element
+                            data['energy'] = energy
+                            data['error'] = 0
+                            data['peak_centre'] = peak
+                            data['transition'] = transition
+                            data['diff'] = 0
+                            data['Rating'] = 4
+                            matches.append(data)
+
+                        elif peak >= (energy - sigma) and peak <= (energy + sigma):
+                            data = {}
+                            data['element'] = element
+                            data['energy'] = energy
+                            data['error'] = sigma
+                            data['peak_centre'] = peak
+                            data['transition'] = transition
+                            data['diff'] = abs(peak - energy)
+                            data['Rating'] = 3
+                            matches.append(data)
+
+                        elif peak >= (energy - (2 * sigma)) and peak <= (energy + (2 * sigma)):
+                            data = {}
+                            data['element'] = element
+                            data['energy'] = energy
+                            data['error'] = 2 * sigma
+                            data['peak_centre'] = peak
+                            data['transition'] = transition
+                            data['diff'] = abs(peak - energy)
+                            data['Rating'] = 2
+                            matches.append(data)
+
+                        elif peak >= (energy - (3 * sigma)) and peak <= (energy + (3 * sigma)):
+                            data['element'] = element
+                            data['energy'] = energy
+                            data['error'] = 3 * sigma
+                            data['peak_centre'] = peak
+                            data['transition'] = transition
+                            data['diff'] = abs(peak - energy)
+                            data['Rating'] = 1
+                            matches.append(data)
+            matches = sorted(matches, key=lambda x: x['diff'])
+            all_matches.append(matches)
+        return all_matches
+
+    def output_data(self,primary_data, secondary_data, all_data,names):
+        prim_table = self.make_peak_table(names[0], primary_data)
```

Comment:
what are the indices of name?

Reply:
what is names?

Reply:
names is a list of name inputted by the user which is used to name the output tables

Reply:
i will change it to something better